pkg/statistics/handle/usage/indexusage/collector.go

```diff
@@ -0,0 +1,225 @@
+// Copyright 2024 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package indexusage
+
+import (
+	"sync"
+	"time"
+
+	"github.com/pingcap/tidb/pkg/parser/model"
+	"github.com/pingcap/tidb/pkg/statistics/handle/usage/collector"
+)
+
+// GlobalIndexID is the key type for indexUsageMap.
+type GlobalIndexID struct {
+	TableID int64
+	IndexID int64
+}
+
+// Sample is the data structure to store index usage information.
+type Sample struct {
+	LastUsedAt     time.Time
+	QueryTotal     uint64
+	KvReqTotal     uint64
+	RowAccessTotal uint64
+	// 0, 0-1, 1-10, 10-20, 20-50, 50-100, 100
+	PercentageAccess [7]uint64
```

Comment:
Please add more comments for those fields.

Reply:
Done